### ISSUE TITLE
Cannot get global or actor folder to work

### ISSUE THREAD

**Issue by ReenigneArcher** [gboudreau/XBMCnfoMoviesImporter.bundle#110]
If I have this in my xml
```
  <actor>
    <name>Actor Name</name>
    <role>Actor Role</role>
  </actor>
```

With the actor thumb location set to global with this path. This path does work and I have website set up to access it. (Also tried using L:\Media\Videos\Actors)
`http://localhost/Media/Videos/Actors`

And the file name "Actor Name.jpg" in the location above.

The agent will not add any image.

Also if I change the setting to local and put the jpg in the same directory as the xml and video file it will not add an image either.

The only option that works is link with the entire path in the xml.

I really want to use the global option if possible. Am I doing something wrong?

Edit: I got the global folder to work by replacing the space with an underscore; however discovered something else. If the image changes then Plex will not fetch a new image even upon refreshing metadata, unmatching/rematching, etc.

**Comment by SlrG** [comment]
Sorry for the late reply. As you already found, a space instead of an underscore was the problem.

Regarding the refreshing, this is a very old plex problem. It caches the data and refuses to update it. Sadly there is nothing the agent can do to change this. (At least not that I know.) The only way is to either delete and recreate the full library or move the movie or series to a different folder, refresh, move it back and refresh it again.

**Comment by dlane2155** [comment]
No worries, not a late reply at all!

Okay, I moved the video entirely out of the library. Emptied the trash. Re-added it, but it kept showing the first image. I will keep messing with it.

And how would I get "local" to work?

**Comment by SlrG** [comment]
Did you refresh the library after moving the video? Then refresh again, after moving it back?

Local should work, if you have the actor jpgs inside a .actors directory inside the videos folder.
Movies/Moviename/.actors

**Comment by BH1NBN** [comment]
> Did you refresh the library after moving the video? Then refresh again, after moving it back?
> 
> Local should work, if you have the actor jpgs inside a .actors directory inside the videos folder.
> Movies/Moviename/.actors

Must the .actors directory exist? I can't leave the actor outside? Because windows cannot create linux [.] Hidden files

**Comment by SlrG** [comment]
Well, my Windows 10 does create the .actors folder without problems. And even .files are possible if you disable the stupid and IMHO dangerous default of hiding the extensions of known file types. 

This stupid setting makes it possible to create virus.txt.exe. It hides the .exe, making the inattentive user click on executables because they are displayed as virus.txt. Also new users don't learn that windows is dependent on the extensions to recognize files. Put on a system without the default setting they delete the extensions when renaming the files or start creating stupid filenames like "Document 1.txt Version 1.txt" because they append after the extension and not before. (Sorry for the rant. :) )

**Comment by BH1NBN** [comment]
> Well, my Windows 10 does create the .actors folder without problems. And even .files are possible if you disable the stupid and IMHO dangerous default of hiding the extensions of known file types.
> 
> This stupid setting makes it possible to create virus.txt.exe. It hides the .exe, making the inattentive user click on executables because they are displayed as virus.txt. Also new users don't learn that windows is dependent on the extensions to recognize files. Put on a system without the default setting they delete the extensions when renaming the files or start creating stupid filenames like "Document 1.txt Version 1.txt" because they append after the extension and not before. (Sorry for the rant. :) )
I didn't make it clear, I really can't create such hidden folders starting with dot on win7

**Comment by ReenigneArcher** [comment]
Think you need to make it not hide system files. It's not the same as hidden files.

**Comment by ReenigneArcher** [comment]
Wanted to add a comment for anyone else having a problem getting actor images from the global folder to update after a change. Here is how I got it to work. Rename actor folder from .actors to .actors1... then change the path in the library settings and refresh all metadata. If another update is made you have to use a name that wasn't used before otherwise it will revert to the old images. So just go .actors2, etc. or some other method as long as the names aren't reused.

**Comment by pimalel233** [comment]
@ReenigneArcher  Hi, I am having issues with setting up my own local/server with actor images, could you share step by step how you made it work with plex? I have tried many ways, it still shows either actor initials or black image. I have even set up a web server that lists all jps under address https://localhost/admin/.actors/, put the link in 'path to movie library or global actor folder' and set to global. Still no luck. I have no idea what I can be missing.

**Comment by ReenigneArcher** [comment]
> @ReenigneArcher Hi, I am having issues with setting up my own local/server with actor images, could you share step by step how you made it work with plex? I have tried many ways, it still shows either actor initials or black image. I have even set up a web server that lists all jps under address https://localhost/admin/.actors/, put the link in 'path to movie library or global actor folder' and set to global. Still no luck. I have no idea what I can be missing.

If you've read this entire issue, I've already described the steps I took to get it working. If you're getting black images instead of letters it seems plex has already cached the image (maybe you used a test image). You'll have to use a new folder (just rename the actor folder). Make sure you have actor images named exactly as they are in the XML files, except replace spaces with underscores.

**Comment by pimalel233** [comment]
@ReenigneArcher I really read that and tested it in many ways, also with the renaming, also tried with deleting the image cache on plex. I even opened up plex database and it did not have save my web server links. The only thing I have not tried yes was putting the local server weblink into nfo but I though it should not be needed with that method. I think I will erase plex and try it fresh, something must be wrong with the installation then. I would not even mind if the 'local' mode worked, which I assume from the thread on plex forums works with images loaded from .actors file within the movie folder, but that did not work also. Appreciate the response.

edit: I would also appreciate sharing currect structure on web server/local how you have those actors set up. Once in logs I had that it was requesting localhost/.actors/moviename/.actors/name_surname.jpg even though I had it set up only localhost/.actors. Then for some next times I could not get it to list that error in logs at all.

**Comment by ReenigneArcher** [comment]
> @ReenigneArcher I really read that and tested it in many ways, also with the renaming, also tried with deleting the image cache on plex. I even opened up plex database and it did not have save my web server links. The only thing I have not tried yes was putting the local server weblink into nfo but I though it should not be needed with that method. I think I will erase plex and try it fresh, something must be wrong with the installation then. I would not even mind if the 'local' mode worked, which I assume from the thread on plex forums works with images loaded from .actors file within the movie folder, but that did not work also. Appreciate the response.
> 
> edit: I would also appreciate sharing currect structure on web server/local how you have those actors set up. Once in logs I had that it was requesting localhost/.actors/moviename/.actors/name_surname.jpg even though I had it set up only localhost/.actors. Then for some next times I could not get it to list that error in logs at all.

You don't need a web server (IIS) setup. You can use a local folder. (C:\ for example).

I think you need to change the option to global actor directory instead of local. Localhost is not the same thing. Localhost in a url is just the same as 127.0.0.1. Using local with get the image from the .actor folder in the movie folder.

I only used a web server because I already have it setup for another purpose. But I wouldn't complicate your setup with that if you don't need it.

Plex downloads the images and caches them, it doesn't fetch them from the source every time.

You can also put a url in the xml file under the Actor/Thumb key.

**Comment by pimalel233** [comment]
@ReenigneArcher I already had an apache running so it was not a problem. I was switching from global/local as I have been trying everything. I really have no idea what is happening. I set up 'local' with http link, enabled debug and it was 'sucessful to set image' with the correct path but the image was black. Then, after refresh metadata again it just kept 'failed setting local actor photo' even on global and then just 'exception setting local or global actor photo!' with python errors. It seems like either plugin is acting up or something else, as even my local subtitles are not recognized without loading plex'' agent which had worked some months ago just fine, before reinstalling plex (leaving the plex data untouched with settings and plugins). It's late for me, so thanks for the help though I do not really see how I will get it working. I guess maybe  total clean plex install and installing plugins from webtools, as the ones I had are the same I used weeks ago that worked. 
Ideally, I would want to keep my .nfo (I do not have xml structure) that is generated by tinyMediaManager that also downloads my .actors. Actors in the nfo have links to the internet but i still would like to load them locally. Even on fresh plex install, no cached content, I am still having issues.

**Comment by pimalel233** [comment]
@ReenigneArcher Finally found some time and managed to get it to work via server after clean install, only in 'local' mode selected. Unfortunately I were not be able to use local path, not sure if it is possible,
Ideally I would get actor images from http://serveraddress/imagelocation/ but seems like it is hardcoded to http://serveraddress/imagelocation/moviename/.actors/  when set to 'http://serveraddress/imagelocation/'. Maybe do you know if is it possible to change it?

**Comment by pimalel233** [comment]
@ReenigneArcher I will be honest with you, I though I had figured it out to use 'local' type, as global just does not work for me, it seems to be using links from the .nfo file instead the path I provide, and in the nfo I have web links which I would like to keep.

With local, I can set path to my local webservers via https because via localhost it's getting an error, probably a certificate for localhost name would fix it. It seems to load the images from the correct path then, example -> https://localhost/data/movies/ but there's a catch. An image loads as empty, actors initials.
So for https://localhost/data/movies/Movie folder name/.actors/name_surname.jpg to load I also need to have X:/Movies/Movie folder name/.actors/name_surname.jpg image with the movie. It can be anything, even wrong one, then my correct picture loads. If I have only local image stored, with nothing on my web link, it loads as blank, no initials. 
Are you able to tell me why it happens this way? It seems like it is an error with the code, if it works for you and you have some free time, I would like to know how. I have a fresh copy of plugin from the https://github.com/gboudreau/XBMCnfoMoviesImporter.bundle/archive/master.zip
Basically it works but only if I have a placeholder of the same name stored locally with the movie.

**Comment by ReenigneArcher** [comment]
Why are you using http or https? Just use the folder path. "C:\my actor folder" Set it to global. Other than that I don't know. I didn't write the plugin. You are free to look at the code though. It's all python.

If initials are loading it's not finding any image.

Nfo links override global folder. If you don't want to use them you need to remove them from the nfo.

**Comment by pimalel233** [comment]
I was using https because it was faster, as http was failing to get the image in the log, mostly because it has no ssl certificalte for the name. Https did so it worked out of the box. The issue there is that it required me to have placefolder of image or image itself in the folder with the movie for it to load the link. otherwise it was blanking out. Strange interaction imo.

For global then, links in my nfo must have been overwriting it, idea was to have them as backup in case local content is not available which I read it works like that, probably for the other method that has another issue unfortunately.

Thanks for clearing up how the global works then, seems like without rewriting part of the plugin my set up will not be possible to make.

**Comment by ReenigneArcher** [comment]
You shouldn't be using http or https. There's no reason to use IIS for this agent unless your actors folder is hosted on another server.

Other than that I can't really help. Maybe create a new issue so the developer of the agent can help you?

**Comment by hairyhippy** [comment]
My actors' folder(s) are stored in the same folder as each movie so how would I set the path for local?  I used Ember Media Manager to scrap movie information, etc and that's how it works in terms of actors... there isn't just one ".actors" folder for all movies.

Edit: the movies are on a NAS so would it be something like http://localhost/volume1/video/movie/ ?  One level below that and there are loads of folders for each individual movie.
